### DISCUSSION TITLE
montage reading filenames from file - Invalid argument @ error/blob.c/OpenBlob/3537

### DISCUSSION THREAD

**Discussion by pkar70** [ImageMagick/ImageMagick#4191] [Help]
### Prerequisites
- [X] I have written a descriptive issue title
- [X] I have verified that I am using the latest version of ImageMagick
- [X] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

 When I try to use this command line (Windows environment):
`dir /s/b *mask* >> %temp%\file.txt`
`magick montage -geometry 320x320^>+1+1 -label "%%t" @%temp%\file.txt sheet.jpg`

I got this response, for every file from %temp%\file.txt:
`montage: unable to open image '@v:L:\film01-Lwow\film01.06\fr12-BS.jpg': Invalid argument @ error/blob.c/OpenBlob/3537`

I tried to convert filenames from absolute to relative, and get message:
`montage: unable to open image '@v:film01.06\fr12-BS.jpg': Invalid argument @ error/blob.c/OpenBlob/3537`
(but `dir film01.06\fr12-BS.jpg` says that this file exist)

I also tried to convert \ to /, but it doesn't change anything (besides changing backslash to forward slash in error message).

When I use only filenames from one dir (not pathnames), message is:
`montage: unable to open image '@v:fr20-BS.jpg': No such file or directory @ error/blob.c/OpenBlob/3537.`

Also (manually) changing new line to space in this file doesn't allow to create contact sheet.  Same error, "no such file".

### Steps to Reproduce

chdir to some directory tree with pictures, then
`dir /s/b *mask* >> %temp%\file.txt`
`magick montage -geometry 320x320^>+1+1 -label "%%t" @%temp%\file.txt sheet.jpg`

### System Configuration

- ImageMagick version: ImageMagick 7.0.10-33 Q16 x64 2020-10-04 
- Environment (Operating system, version and so on): Microsoft Windows [Version 10.0.19041.508]
- Additional information:

**Comment by snibgo** [comment]
ImageMagick doesn't like Windows drive names (like "L:") in filenames, and especially not when those filenames are in file lists, such as you have here.

"dir /s/b *mask*" will include drive names. Try just "dir /b *mask*", to omit the drive (and directory). Does that work? If so, that was the problem.

If you don't need "/s" (for subdirectories) then this is the solution.

If you do need "/s", I might have some other ideas.

**Comment by pkar70** [comment]
As I wrote in opening issue:
I tried to convert filenames from absolute to relative, and get message:
`montage: unable to open image '@v:film01.06\fr12-BS.jpg': Invalid argument @ error/blob.c/OpenBlob/3537`
(but dir film01.06\fr12-BS.jpg says that this file exist)

filename in file was:
`film01.06\fr12-BS.jpg`

(and some others, each in separate line)

And, also from opening issue:
`montage: unable to open image '@v:fr20-BS.jpg': No such file or directory @ error/blob.c/OpenBlob/3537` - only filenames, without any directories.

**Comment by snibgo** [comment]
> montage: unable to open image '@v:film01.06\fr12-BS.jpg'

I don't know where "@v:" has come from.

**Comment by pkar70** [comment]
I know :)
I thought that it was something like from method parameter name, but... it was drive with file given after @. (in real scenario, I use `dir/s/b > %temp%\makejpgcs.txt`, and `magick ... @%temp%\makejpgcs.txt` (and %temp% is `v:\UserTemp\pkar` ). So, I think problem is, why @file logic prefixes filenames from @file with drive with this @file?

When @file is within dirtree with pictures (same drive), I encounter another problem - spaces within filenames.  How to give file name with spaces within @file?
